server/routerlicious/packages/services-client/src/heap.ts

```diff
@@ -0,0 +1,84 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+export interface IHeapComparator<T> {
+	compareFn(a: T, b: T): number;
+}
+
+export class Heap<T> {
```

Comment:
I think the `mergeKArrays` tests technically tests `Heap`, but some UTs for this new class would be awesome if you have time.